### PR TITLE
We were missing some critical files and directories in monitoring, th…

### PR DESCRIPTION
**Pull request by mpurzynski**
…is is the endgame.

### CODE REVIEW

File: example_audit.rules [hunk 1]
```diff
@@ -88,20 +88,46 @@ w /etc/fluent/ -p wa -k syslog
...
+-w /etc/at.deny -p wa -k cron
+-w /etc/anacrontab -p wa -k cron
  
 # Modify startup
+-w /boot -p wa -k init
 -w /etc/rc.d/init.d/ -p wa -k init
 -w /sbin/init -p wa -k init
 -w /etc/inittab -p wa -k init
 -w /etc/systemd/ -p wa -k init
+-w /etc/systemd/system -p wa -k init
```
Comment (gdestuynder) [inline example_audit.rules]: /etc/systemd should be recursive already

File: example_audit.rules [hunk 2]
```diff
@@ -88,20 +88,46 @@ w /etc/fluent/ -p wa -k syslog
...
+-w /usr/lib/systemd/system -p wa -k init
 -w /etc/init.d/ -p wa -k init
+-w /etc/rc0.d -p wa -k init
+-w /etc/rc1.d -p wa -k init
+-w /etc/rc2.d -p wa -k init
+-w /etc/rc3.d -p wa -k init
+-w /etc/rc4.d -p wa -k init
+-w /etc/rc5.d -p wa -k init
+-w /etc/rc6.d -p wa -k init
+-w /etc/rc.d -p wa -k init
```
Comment (gdestuynder) [inline example_audit.rules]: /etc/rc.d should be recursive already
also given the amount of entries, should this just be watching `/etc` ? ;-)
**each** entry slows down syscall entry latency for writes and attribute change to any file

File: example_audit.rules [hunk 3]
```diff
@@ -157,19 +183,32 @@ w /etc/fluent/ -p wa -k syslog
...
+-w /etc/modprobe.conf.local -p wa -k kernel
+-w /etc/modprobe.d/ -p wa -k kernel
+-w /etc/modules-load.d/ -p wa -k kernel
 
 # Kernel modules
--w /lib/modules/ -p wa -k binaries
+-w /lib/modules/ -p wa -k kernel
+
+# Critical devices
+-w /dev/mem -p wa -k kernel
```
Comment (gdestuynder) [inline example_audit.rules]: mknode /dev/other-name-here ;-)
is this really that useful?

File: example_audit.rules [hunk 4]
```diff
@@ -157,19 +183,32 @@ w /etc/fluent/ -p wa -k syslog
...
+-w /dev/cpu/1/msr -p wa -k kernel
+-w /dev/cpu/3/msr -p wa -k kernel
+-w /dev/cpu/4/msr -p wa -k kernel
+-w /sys/kernel/debug -p wa -k kernel
 
 # Libc preload and linking configuration
 -w /etc/ld.so.conf -p wa -k binaries
 -w /etc/ld.so.conf.d -p wa -k binaries
 -w /etc/ld.so.preload -p wa -k binaries
+-w /lib/x86_64-linux-gnu/ld-2.19.so -p wa -k binaries
```
Comment (gdestuynder) [inline example_audit.rules]: that seems very specific (too specific)
if i can write here ill just change the libc name

File: example_audit.rules (hunk 1 above)
Comment (mpurzynski) [reply example_audit.rules]: Removed

File: example_audit.rules (hunk 2 above)
Comment (mpurzynski) [reply example_audit.rules]: Removed

File: example_audit.rules (hunk 4 above)
Comment (mpurzynski) [reply example_audit.rules]: This one is included in the /lib monitoring

File: example_audit.rules (hunk 3 above)
Comment (mpurzynski) [reply example_audit.rules]: Indeed - removed. kmem is disabled by default on any modern distribution, mem gives read-only access and if someone has privileges to access ports/iopl/ioperm/msr they can just backdoor the kernel.

**Pull request merged**